lib/upperkut/strategies/buffered_queue.rb

```diff
@@ -7,13 +7,65 @@ module Strategies
     class BufferedQueue < Upperkut::Strategies::Base
       include Upperkut::Util
 
+      DEQUEUE_ITEMS = %(
+        local key = KEYS[1]
+        local waiting_ack_key = KEYS[2]
+        local batch_size = ARGV[1]
+        local current_timestamp = ARGV[2]
+        local expired_ack_timestamp = ARGV[3] + 1
+
+        -- move expired items back to the queue
+        local expired_ack_items = redis.call("ZRANGEBYSCORE", waiting_ack_key, 0, expired_ack_timestamp)
+        if table.getn(expired_ack_items) > 0 then
+          redis.call("ZREMRANGEBYSCORE", waiting_ack_key, 0, expired_ack_timestamp)
+          for i, item in ipairs(expired_ack_items) do
+            redis.call("RPUSH", key, item)
```

Comment:
`RPUSH` is a variadic function which can receive multiple items, can we do it here instead of multiple calls?
https://redis.io/commands/rpush

Reply:
I've tried to splat an items array to use the variadic interface but didn't manage to make it work, so as the number of operations was greatly reduced by using `LRANGE` + `LTRIM`, I thought about giving it a try and work on improvement.

What do you think?

Reply:
LGTM